src/mpi/datatype/typeutil.c

```diff
@@ -564,5 +565,22 @@ void MPIR_Datatype_free(MPIR_Datatype * ptr)
     if (ptr->typerep) {
         MPIR_Typerep_free(&(ptr->typerep));
     }
+    MPL_free(ptr->flattened);
```

Comment:
I think this needs to be protected with a NULL check?

Reply:
`libc` allows freeing `NULL`. Apparently, freeing without checking is the MPICH style ... and I hated it. There is no balance reading the code.

Reply:
But `MPL_free` can also go into our own `trmem` allocator, depending on the configure options.

Reply:
`MPL_trfree` checks `NULL` ptr too.

Reply:
Alright, I withdraw my objection then.